src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,199 @@
+            ResultSet carsResultSet = getAllCarsStatement.executeQuery();
+            while (carsResultSet.next()) {
+                cars.add(getCar(carsResultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t get all cars ", e);
+        }
+        for (Car car : cars) {
+            car.setDrivers(getDriversFromDataBase(car.getId()));
+        }
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String updateCarRequest = "UPDATE cars SET model = ?, manufacturer_id = ? "
+                + "WHERE id = ? AND is_deleted = 'false'";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateCarStatement
+                        = connection.prepareStatement(updateCarRequest)) {
+            updateCarStatement.setString(1, car.getModel());
+            updateCarStatement.setLong(2, car.getManufacturer().getId());
+            updateCarStatement.setLong(3, car.getId());
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t update car " + car, e);
+        }
+        deleteDriversFromDataBase(car);
+        insertDriversToDataBase(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteRequest = "UPDATE cars SET is_deleted = 'true' WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteStatement = connection.prepareStatement(deleteRequest)) {
+            deleteStatement.setLong(1, id);
+            return deleteStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t delete car by id" + id, e);
+        }
+    }
+
+    public List<Car> getAllByDriver(Long driverId) {
+        List<Long> carsId = new ArrayList<>();
+        List<Car> cars = new ArrayList<>();
+        String getAllByDriverRequest
+                = "SELECT car_id, model, manufacturer_id, manufacturers.name, country "
+                + "FROM cars_drivers INNER JOIN cars ON car_id = cars.id "
+                + "INNER JOIN drivers ON driver_id = drivers.id "
+                + "INNER JOIN manufacturers ON manufacturer_id = manufacturers.id "
+                + "WHERE driver_id = ? "
+                + "AND drivers.is_deleted = 'false' AND cars.is_deleted = 'false';";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllByDriverStatement =
+                        connection.prepareStatement(getAllByDriverRequest)) {
+            getAllByDriverStatement.setLong(1, driverId);
+            ResultSet getAllByDriverResultSet = getAllByDriverStatement.executeQuery();
+            while (getAllByDriverResultSet.next()) {
+                carsId.add(getAllByDriverResultSet.getLong("car_id"));
```

Comment:
I think we should get all info about car and it's manufacturer in this request, not just car ids